nix/installer.nix

```diff
@@ -11,17 +11,66 @@
   }).latestCompatibleLinuxPackages;
   boot.zfs.removeLinuxDRM = lib.mkDefault pkgs.hostPlatform.isAarch64;
 
-  documentation.enable = false;
+  documentation.enable = lib.mkForce false;
+
+  # reduce closure size through package set crafting
+  # where there's no otherwise globally effective
+  # config setting available
+  # TODO: some are candidates for a long-term upstream solution
+  nixpkgs.overlays = [
+    (final: prev: {
+      # save ~12MB by not bundling manpages
+      coreutils-full = prev.coreutils;
+      # save ~20MB by making them minimal
+      util-linux = prev.util-linux.override {
+        nlsSupport = false;
+        ncursesSupport = false;
+        systemdSupport = false;
+        translateManpages = false;
+      };
+      # save ~6MB by removing one bash 
+      bashInteractive = prev.bash;
+      # saves ~25MB
+      systemd = prev.systemd.override {
+        pname = "systemd-slim";
+        withDocumentation = false;
+        withCoredump = false;
+        withFido2 = false;
+        withRepart = false;
+        withMachined = false;
+        withRemote = false;
+        withTpm2Tss = false;
+        withLibBPF = false;
+        withAudit = false;
+        withCompression = false;
+        withImportd = false;
+        withPortabled = false;
+        withSysupdate = false;
+        withHomed = false;
+        withLocaled = false;
+        withPolkit = false;
+        # withQrencode = false;
+        # withVmspawn = false;
+        withPasswordQuality = false;
+      };
+    })
+  ];
+  systemd.coredump.enable = false;
+
 
   environment.systemPackages = [
     # for zapping of disko
     pkgs.jq
     # for copying extra files of nixos-anywhere
     pkgs.rsync
+    # for installing nixos via nixos-anywhere
+    config.system.build.nixos-enter
+    config.system.build.nixos-install
   ];
 
   imports = [
     ./nix-settings.nix
+    ./un-pearl.nix
```

Comment:
We still have switch-to-configuration.pl after that.

Reply:
That's true.

Maybe I did ingest that wrongly, however my current understanding was that `nixos-anywhere` at "most" requires `nixos-install` which depends on `nixos-enter` and if that depends on `switch-to-...`, then it indeed escaped my attention.

Reply:
No I mean. There is also a switch-to-configuration that the system itself has independent on those nixos tools.

Reply:
Ah, I guess that was weeded out by `system.switch.enable = false;`